corehq/apps/accounting/invoicing.py

```diff
@@ -353,6 +327,165 @@ def create_wire_credits_invoice(self, items, amount):
+        if subscription.subscriptionadjustment_set.count() == 0:
+            SubscriptionAdjustment.record_adjustment(
+                subscription,
+                method=SubscriptionAdjustmentMethod.TASK,
+                invoice=invoice,
+            )
+        generate_line_items(invoice, subscription)
+        invoice.calculate_credit_adjustments()
+        invoice.update_balance()
+        invoice.save()
+        visible_domain_invoices = Invoice.objects.filter(
+            is_hidden=False,
+            subscription__subscriber__domain=invoice.get_domain(),
+        )
+        total_balance = sum(invoice.balance for invoice in visible_domain_invoices)
+        should_set_date_due = (
+            total_balance > SMALL_INVOICE_THRESHOLD or
+            (invoice.account.auto_pay_enabled and total_balance > Decimal(0))
+        )
+        if should_set_date_due:
+            days_until_due = DEFAULT_DAYS_UNTIL_DUE
+            if subscription.date_delay_invoicing is not None:
+                td = subscription.date_delay_invoicing - self.date_end
+                days_until_due = max(days_until_due, td.days)
+            invoice.date_due = self.date_end + datetime.timedelta(days_until_due)
+        invoice.save()
+        return invoice
+
+    def _consolidate_invoices(self, invoices):
+        customer_invoice = invoices.pop()
+        invoiced_plans = [customer_invoice.subscription.plan_version]
+        for invoice in invoices:
+            if invoice.subscription.plan_version in invoiced_plans:
+                # The plan has already been added to the invoice, so update the line items
+                for line_item in invoice.lineitem_set.all():
+                    if line_item.product_rate:
+                        continue
+                    else:
+                        self._update_line_item_to_include(line_item)
+            else:
+                line_item_set = customer_invoice.lineitem_set.union(invoice.lineitem_set.all(), all=True)
+                customer_invoice.lineitem_set.set(line_item_set)
+                customer_invoice.save()
+        customer_invoice.calculate_credit_adjustments()
+        customer_invoice.update_balance()
+        customer_invoice.save()
+        should_set_date_due = (
+            customer_invoice.balance > SMALL_INVOICE_THRESHOLD or
+            (customer_invoice.account.auto_pay_enabled and customer_invoice.balance > Decimal(0))
+        )
+        if should_set_date_due:
+            customer_invoice.date_due = self.date_end + datetime.timedelta(DEFAULT_DAYS_UNTIL_DUE)
+        customer_invoice.save()
+        return customer_invoice
+
+    def _update_line_item_to_include(self, line_item):
+        try:
+            line_item_to_update = self.customer_invoice.lineitem_set.get(feature_rate=line_item.feature_rate)
+            line_item_to_update.quantity = line_item_to_update.quantity + line_item.quantity
+        except ObjectDoesNotExist:
```

Comment:
It's generally preferable to use ```ModelName.DoesNotExist```.

Example: https://github.com/dimagi/commcare-hq/blob/088b2f808827a70e106bc54d4adbcab383019f49/corehq/apps/accounting/views.py#L158